Core/Src/leds/rgb_led_handler.c

```diff
@@ -39,32 +35,30 @@ void rgb_led(void* argument) {
     for (;;) {
         ECU_ENABLE_BREAKPOINT_DEBUG();
 
-        // espera RTD ser setado ou timeout estourar
-        switch (
-            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_BLINK_DELAY)) {
+        // wait until the RTD is set or to timeout overflow
 
-            // caso timeout estore vai piscar o led, indicando que ta fora do RTD
-            case osErrorTimeout:
-                write_rgb_color(get_rgb_color(message.color));
-                blink_rgb(RGB_BLINK_DELAY);
-                break;
+        switch (osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_TIMEOUT)) {
+
+            // if the timeout overflows the led will blink to show the car its not in RTD
+            case osErrorTimeout: write_pattern(message, RGB_BLINK_200_DELAY); break;
 
             default:
-                switch (message.control) {
-                    case FIXED:
-                        for (;;) {
-                            write_rgb_color(get_rgb_color(message.color));
-                            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL,
-                                              osWaitForever);
-                            if (message.control == BLINK200) {
-                                break;
-                            }
+                if (message.control == FIXED) {
+                    if (message.sizeOfPattern > 1) {
+                        while (!are_messages_available(q_rgb_led_messageHandle)) {
+                            write_pattern(message, RGB_BLINK_1000_DELAY);
                         }
                         break;
-                    default:
-                        write_rgb_color(get_rgb_color(message.color));
-                        blink_rgb(RGB_BLINK_DELAY);
-                        break;
+                    }
+                    write_rgb_color(
+                        get_rgb_color(message.pattern[ONE_COLOR_PATTERN_POS]));
+                    while (!are_messages_available(q_rgb_led_messageHandle)) {
+                        osDelay(RGB_BLINK_200_DELAY);
+                    }
+                }
+
+                else if (message.control == BLINK200) {
+                    write_pattern(message, RGB_BLINK_200_DELAY);
```

Comment:
Essa parte tá confusa. O que o fixed e blink significa agora? pq ele pisca no fixed tbm. tenta fazer como falei no outro comentário de fazer um pisca temporário.  A parte do timeout tbm acho que tá redundante.

Reply:
O "fixed" fixa, se for um padrão fixa o padrão no LED, que se repete até q não seja mais necessário, se for uma cor ela fica fixa sem mudar